### PR TITLE
Change php7 errors to exception to maintain compatability with function declaration #264

### PR DESCRIPTION
**Pull request by jamiehd**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     "require": {
         "php": ">=5.3.0",
         "psr/log": "^1.0",
-        "symfony/var-dumper": "^2.6|^3.0"
+        "symfony/var-dumper": "^2.6|^3.0",
+        "symfony/debug": "^2.6|^3.0"
```
Comment (GrahamCampbell) [inline composer.json]: should be sorted alphabetically

File: src/DebugBar/DataCollector/ExceptionsCollector.php [hunk 2]
```diff
@@ -29,6 +31,9 @@ public function addException(Exception $e)
     {
         $this->exceptions[] = $e;
         if ($this->chainExceptions && $previous = $e->getPrevious()) {
+            if ($previous instanceof Error) {
```
Comment (GrahamCampbell) [inline src/DebugBar/DataCollector/ExceptionsCollector.php]: `!$previous instanceof Exception` would be more correct

Comment (barryvdh) [inline src/DebugBar/DataCollector/ExceptionsCollector.php]: I disagree, I think this is more readable and more explicitly states the purpose of the code.

Comment (barryvdh) [inline src/DebugBar/DataCollector/ExceptionsCollector.php]: And while https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Debug/Exception/FatalThrowableError.php accepts Throwable, it seems to be explicitly aimed at the Error class.

**Comment by barryvdh** [comment]
Thanks!

**Comment by GrahamCampbell** [comment]
@barryvdh I don't agree. Throwables don't have to extend Error, and at no point does symfony every typehint `Error`, for that exact reason.

**Comment by GrahamCampbell** [comment]
Something can be throwable, but not be an error or an exception.

**Comment by barryvdh** [comment]
What then?

**Comment by GrahamCampbell** [comment]
I was thinking about forwards compatibility. Currently, PHP has hacked this to bits so that we'd end up with:

```
Fatal error: Class Foo cannot implement interface Throwable, extend Exception or Error instead
```

**Comment by barryvdh** [comment]
Hmm, fair enough. Submit a PR and I'll accept.

**Comment by barryvdh** [comment]
An alternative would be to add `addThrowable` method, because there internally isn't a reason we couldn't just use the Throwables directly. We just can't change the addException typehint.

**Pull request merged**